### PR TITLE
DOCSP-45736-add-self-managed-to-glossary

### PR DESCRIPTION
**Pull request by jeff-allen-mongo**
# Pull Request Process

Please read the
[Style Guide Review Process](https://wiki.corp.mongodb.com/display/DE/Technical+Writer+Style+Guide+Review+Process)
wiki page.

Contributors should take the following actions:

- Request a PR review in #docs-style-guide Slack channel.
- Add the Style Guide team reps as reviewers.
- If any reviewer requests changes, address them and request another review.
- When you receive PR review approvals from all the Style Guide team reps,
  they will merge the PR and notify you of the merge in the Slack channel.

# Pull Request Description

This PR adds a glossary entry for "self-managed deployments", used to refer to non-Atlas deployments. The goal of this PR is to standardize our terminology for non-Atlas deployments, since we currently use a mix of "self-managed", "self-hosted", and "on-premises".

The justification for "self-manged" over other terms is that this is the term proposed by Sarah Lin, and is the term we've been using as part of ongoing efforts in the manual to separate self-hosted and general MongoDB content.

JIRA URL: https://jira.mongodb.org/browse/DOCSP-45736

Staging:

- [List of Terms - O](https://deploy-preview-182--mongodb-docs-meta.netlify.app/style-guide/terminology/alphabetical-terms/#o) 
- [List of Terms - S](https://deploy-preview-182--mongodb-docs-meta.netlify.app/style-guide/terminology/alphabetical-terms/#s)

### CODE REVIEW

File: source/style-guide/includes/terms/o.rst [hunk 1]
```diff
@@ -49,10 +49,15 @@ O
...
-     through the cloud). Always hyphenate. *Premises* is plural;
+
+     :icon-fa4:`times-circle` Avoid when referring to non-Atlas
+     deployments. To refer to non-Atlas deployments, use
+     :term:`self-managed`.
+
+     When used in other contexts, use *on-premises* to distinguish local
+     computing (in which resources are located on a customer's own site)
+     from remote computing (in which resources are provided partially or
+     totally through the cloud). Always hyphenate. *Premises* is plural;
```
Comment (corryroot) [inline source/style-guide/includes/terms/o.rst:60]:
```suggestion
     :icon-fa4:`times-circle` Avoid this term when you refer to non-Atlas
     deployments. To refer to non-Atlas deployments, use
     :term:`self-managed`.

     When you use this term in other contexts, use *on-premises* to distinguish local
     computing (in which resources are located on a customer's own site)
     from remote computing (in which resources are provided partially or
     totally through the cloud). Always hyphenate. *Premises* is plural;
```

File: source/style-guide/includes/terms/s.rst [hunk 2]
```diff
@@ -69,6 +69,11 @@ S
      Usually words with the prefix *self*, such as *self-service* and
      *self-explanatory*, are hyphenated.
 
+   self-managed
+     Use to describe deployments that are not hosted on MongoDB Atlas.
+     Self-managed deployments include deployments that are hosted
+     on-premises as well as non-Atlas environments like AWS.
```
Comment (corryroot) [inline source/style-guide/includes/terms/s.rst:75]:
```suggestion
     Use this term to describe deployments that are not hosted on MongoDB Atlas.
     Self-managed deployments include deployments that are hosted
     on-premises as well as non-Atlas environments like AWS.
```

**Review by corryroot — COMMENTED** [review]
Thanks @jeff-allen-mongo! I commented with some minor copy nits.

File: source/style-guide/includes/terms/s.rst (hunk 2 above)
Comment (jeff-allen-mongo) [reply source/style-guide/includes/terms/s.rst:75]: Updated

File: source/style-guide/includes/terms/o.rst (hunk 1 above)
Comment (jeff-allen-mongo) [reply source/style-guide/includes/terms/o.rst:60]: Updated with a few wording tweaks.

**Review by corryroot — APPROVED** [review]
LGTM!

File: source/style-guide/includes/terms/o.rst [hunk 3]
```diff
@@ -49,11 +49,16 @@ O
...
+     non-Atlas deployments. To distinguish non-Atlas deployments, use
+     :term:`self-managed`.
+
+     Use *on-premises* to distinguish local computing (in which
+     resources are located on a customer's own site) from remote
      computing (in which resources are provided partially or totally
-     through the cloud). Always hyphenate. *Premises* is plural;
-     don't use *on-premise* or *off-premise*.
+     through the cloud). Always hyphenate. *Premises* is plural; don't
+     use *on-premise* or *off-premise*.
```
Comment (stephmarie17) [inline source/style-guide/includes/terms/o.rst:61]: [q] Not sure how prevalent it is here, but in past companies I've also seen `on-prem` used pretty frequently - is it worth calling that out here or does this cover it?

**Review by stephmarie17 — APPROVED** [review]
Thank you for adding ✨ LGTM with a small comment

File: source/style-guide/includes/terms/o.rst (hunk 3 above)
Comment (jeff-allen-mongo) [reply source/style-guide/includes/terms/o.rst]: @stephmarie17 thanks for this comment. I added a small clarification that for consistency we should use "on-premises". Let me know if you'd like to see anything different here.

Comment (stephmarie17) [reply source/style-guide/includes/terms/o.rst]: LGTM 👍

**Review by lindseymoore — APPROVED** [review]
LGTM, thanks!

**Pull request merged**